src/effect/delay.rs

```diff
@@ -74,6 +85,13 @@ impl Delay {
         Some(new_value)
     }
 
+    fn delay_from_midi_messages_tap(&mut self, messages: &[midi::Message]) -> Option<u32> {
+        let tap_tempo = self.tap_tempo.as_mut()?;
```

Comment:
If you have a mutable reference to an `Option`, calling `as_mut` will make it so the data it contains (if it is `Some`) is mutable to you as well, which I need in the next line because `handle_messages` requires a mutable self.